src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,127 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFileAsSpan(statusFilePath, reusableReader, out ReadOnlySpan<char> statusFileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var results = default(ParsedStatus);
+            while (!statusFileContents.IsEmpty)
+            {
+                int startIndex = statusFileContents.IndexOf(':');
+                int endIndex = statusFileContents.IndexOf('\n');
+
+                // Reached end of file
+                if (endIndex == -1)
+                {
+                    break;
+                }
+
+                ReadOnlySpan<char> title = statusFileContents.Slice(0, startIndex);
+                ReadOnlySpan<char> value = default;
+                // endIndex - startIndex - 1 --> To slice till just before '\n'
+                // endIndex - startIndex - 4 --> To slice till just before ' kB\n'
+                switch (title.ToString())
+                {
+#if DEBUG
+                    case "Pid":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 1);
```

Comment:
Calculating startIndex+1 and endIndex-startIndex-1 is cheap: I suggest doing it above once rather than copying this code for each branch.  In fact, the slice is cheap, too: just move this whole line to before the switch above.  Paying for it even when the field doesn't match won't be meaningful overhead, and the code will be smaller / easier to maintain.

Reply:
Cannot move the whole line but only the index calculation as if I slice, chances are that for some fields like `Name:   bash`, it will throw an exception if try slicing with `endIndex - startIndex - 4` by default due to negative length (8-4-4 = -4 in this case).

Reply:
I don't understand.  What would the problem be for "Name: bash"?  startIndex would be 4, endIndex would be 10, statusFileContents.Slice(4 + 1, 10 - 4 - 1) would be statusFileContents.Slice(5, 5), which would give you back " bash".

Reply:
`10 - 4 - 1`. The `-1` is there when there is no unit at the end. When there is a whitespace trailing `kb` it then comes to `-4`. All lines have different lengths and units. 

```
Name:   bash // Here I do -1 to trim only newline char, 
VmPeak:     131168 kB // Here I do -4 to trim newline as well as kb followed by whitespace
```

Hence, cannot blindly slice of values.

Reply:
Implemented as discussed.